server/commitlog/commitlog.go

```diff
@@ -225,7 +230,7 @@ func (l *commitLog) Append(msgs []*Message) ([]int64, error) {
 		segment          = l.activeSegment()
 		basePosition     = segment.Position()
 		baseOffset       = segment.NextOffset()
-		ms, entries, err = newMessageSetFromProto(baseOffset, basePosition, msgs)
+		ms, entries, err = newMessageSetFromProto(baseOffset, basePosition, msgs, l.IsConcurrencyControlEnabled())
```

Comment:
I need to think about this a bit. There is an important caveat here in that `msgs` represents a batch of messages, potentially from entirely separate publishers. But even in the case of them being from the same publisher, there is technically an edge case where some of the messages set an expected offset and some don't (`-1`). In this case, an OCC rejection could cause the rejection of a legitimate message since the entire batch is rejected.

Reply:
After further reading of the code, it looks like you've addressed this by adjusting the batch size to 1. Well done.